packages/flutter_tools/test/general.shard/base/deferred_component_test.dart

```diff
@@ -56,10 +53,10 @@ void main() {
       component.assignLoadingUnits(loadingUnits1);
 
       expect(component.assigned, true);
-      expect(component.loadingUnits.length, 2);
-      expect(component.loadingUnits.contains(loadingUnits1[0]), true);
-      expect(component.loadingUnits.contains(loadingUnits1[1]), true);
-      expect(component.loadingUnits.contains(loadingUnits1[2]), false);
+      expect(component.loadingUnits!.length, 2);
```

Comment:
You could avoid the `!` by writing these as:

`expect(component.loadingUnits, contains(loadingUnits1[0]));`

Reply:
And `hasLength`.  Done.